apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js

```diff
@@ -226,6 +226,47 @@ export const getStudentMCResponsesForCurrentAssessment = (state) => {
   return studentResponsesArray;
 };
 
+/**
+ * Returns an array of objects, each representing a free response question
+ * and all the students' responses to that question.
+ */
+export const getAssessmentsFreeResponseResults = (state) => {
+  const assessmentsStructure = getCurrentAssessmentQuestions(state);
+  if (!assessmentsStructure) {
+    return [];
+  }
+
+  // Initialize an object for each question, with the questionText and
+  // an empty array of responses.
+  const questionData = assessmentsStructure.questions;
+  const questionsAndResults = questionData.filter(question => question.type === 'FreeResponse').map(question => {
+    return {
+      questionText: question.question_text,
+      responses: [],
+    };
+  });
+
+  const studentResponses = getAssessmentResponsesForCurrentScript(state);
+
+  // For each student, look up their responses to the currently selected assessment.
+  Object.keys(studentResponses).forEach(studentId => {
+    studentId = parseInt(studentId);
```

Comment:
Best practices: Always provide the radix parameter (`parseInt(studentId, 10)`) when using [`parseInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt), unless you actually want to support hexadecimal and octal input!  Without it, any string with a leading zero will be treated as octal.

Reply:
Thanks! This is a good point.  When will you want to support both hexadecimal and octal inputs?

Reply:
In our case, rarely.  Maybe we're building a tool that lets students enter red, green, and blue values to define a color, and we want to allow students to type `255` or `0xFF` because they may be familiar with hexadecimal in a color context.  But it will be special cases - I think the answer is, "you'll know."